polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaInterpreter.scala

```diff
@@ -30,15 +30,7 @@ class ScalaInterpreter(
   import kernelContext.{global, runtimeMirror, runtimeTools, importFromRuntime, importToRuntime}
   private val logger = getLogger
 
-  private val executor = Executors.newCachedThreadPool(new ThreadFactory {
-    def newThread(r: Runnable): Thread = {
-      val thread = new Thread(r)
-      thread.setContextClassLoader(kernelContext.classLoader)
-      thread
-    }
-  })
-
-  protected implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.fromExecutorService(executor))
+  protected implicit val contextShift: ContextShift[IO] = IO.contextShift(kernelContext.executionContext)
```

Comment:
TODO: I think we can probably eliminate a lot of one-off thread pools and just have this one thread pool live in the `KernelContext`. It turns out to be very important to always set the context classloader, so it's a good idea to do it all in one place.

Maybe even define the `contextShift` itself in `KernelContext`.